src/numbers/scint.js

```diff
@@ -0,0 +1,118 @@
+import { XdrInt } from './xdrint';
+
+export { XdrInt };
+
+/**
+ * Provides an easier way to manipulate large numbers for Stellar operations.
+ *
+ * You can instantiate this value either from bigints, strings, or numbers
+ * (whole numbers, or this will throw).
+ *
+ * If you need to create a native BigInt from a list of integer "parts" (for
+ * example, you have a series of encoded 32-bit integers that represent a larger
+ * value), you should use the specific XDR type, like {@link U128}. For example,
+ * you could do `new U128(values...).toBigInt()`.
+ *
+ * @example
+ * ```js
+ * import sdk from "stellar-base";
+ *
+ * // You have an ScVal from a contract and want to parse it into JS native.
+ * const value = sdk.xdr.ScVal.fromXDR(someXdr, "base64");
+ * const bigi = sdk.ScInt.fromScVal(value); // grab it as a BigInt
+ * let sci = new ScInt(bigi);
+ *
+ * sci.toNumber(); // gives native JS type (w/ size check)
+ * sci.toBigInt(); // gives the native BigInt value
+ * sci.toU64();    // gives ScValType-specific XDR constructs (with size checks)
+ *
+ * // You have a number and want to shove it into a contract.
+ * sci = sdk.ScInt(0xdeadcafebabe);
+ * sci.toBigInt() // returns 244838016400062n
+ * sci.toNumber() // throws: too large
+ *
+ * // Pass any to e.g. a Contract.call(), conversion happens automatically
+ * // regardless of the initial type.
+ * const scValU128 = sci.toU128();
+ * const scValI256 = sci.toI256();
+ * const scValU64  = sci.toU64();
+ *
+ * // Lots of ways to initialize:
+ * sdk.ScInt("123456789123456789")
+ * sdk.ScInt(123456789123456789n);
+ * sdk.ScInt(1n << 140n);
+ * sdk.ScInt(-42);
+ * sdk.ScInt.fromScVal(scValU128); // from above
+ *
+ * // If you know the type ahead of time (accessing `.raw` is faster than
+ * // conversions), you can specify the type directly (otherwise, it's
+ * // interpreted from the numbers you pass in):
+ * const i = sdk.ScInt(123456789n, { type: "u256" });
+ *
+ * // For example, you can use the underlying `sdk.U256` and convert it to an
+ * // `xdr.ScVal` directly like so:
+ * const scv = new xdr.ScVal.scvU256(i.raw);
+ *
+ * // Or reinterpret it as a different type (size permitting):
+ * const scv = i.toI64();
+ * ```
+ *
+ * @param {number|bigint|string|ScInt} value - a single, integer-like value
+ *    which will be interpreted in the smallest appropriate XDR type supported
+ *    by Stellar (64, 128, or 256 bit integer values). signed values are
+ *    supported, though they are sanity-checked against `opts.type`. if you need
+ *    32-bit values, you can construct them directly without needing this
+ *    wrapper, e.g. `xdr.ScVal.scvU32(1234)`.
```

Comment:
Suggestion: Would it be convenient to also support 32-bit ints here as well? You can make them directly, but it might be more convenient to just have one way to do all ints. Fewer exceptions to remember.

Reply:
It's a little obnoxious, unfortunately, because then the underlying `.int` is not a `jsXdr.LargeInt` type, which provides the essential `.size`, `.toBigInt()`, etc. methods. We could upscale it to an `Int64`, but that feels worse than ignoring the parameter to me. Wdyt?